src/main/java/seedu/address/model/recipe/Recipe.java

```diff
@@ -0,0 +1,130 @@
+     */
+    public Recipe(Name name, List<Ingredient> ingredientList, List<RecipeStep> recipeSteps) {
+        requireAllNonNull(name, ingredientList, recipeSteps);
+        this.uuid = new UniqueId();
+        this.name = name;
+        this.ingredientList = ingredientList;
+        this.recipeSteps = recipeSteps;
+    }
+
+    /** Creates a new Recipe with the specified {@code id}. */
+    public Recipe(int id, Name name, List<Ingredient> ingredientList, List<RecipeStep> recipeSteps) {
+        requireAllNonNull(id, name, ingredientList, recipeSteps);
+        this.uuid = new UniqueId(id);
+        this.name = name;
+        this.ingredientList = ingredientList;
+        this.recipeSteps = recipeSteps;
+    }
+
+    public Name getName() {
+        return this.name;
+    }
+
+    public int getId() {
+        return this.uuid.getId();
+    }
+
+    public List<Ingredient> getIngredients() {
+        return this.ingredientList;
+    }
+
+    public List<String> getRecipeSteps() {
+        return this.recipeSteps.stream().map(RecipeStep::toString).collect(Collectors.toList());
+    }
+
+    /** Check if the current recipe contains the specified ingredient name. */
+    public boolean containsIngredient(Name ingredientName) {
+        requireNonNull(ingredientName);
+        return ingredientList.stream().anyMatch(x -> x.getName().equals(ingredientName));
+    }
+
+    /**
+     * Changes the specified recipe step with the new instructions.
+     */
+    public Recipe modifyRecipeStep(int stepNumber, String newStep) {
+        requireNonNull(newStep);
+        if (stepNumber > recipeSteps.size()) {
+            throw new IllegalArgumentException("Specified step number cannot exceed the current steps of recipe");
+        }
+        List<RecipeStep> stepCopy = new ArrayList<>();
+        stepCopy.addAll(this.recipeSteps);
+        stepCopy.set(stepNumber - 1, stepCopy.get(stepNumber - 1).modifyStep(newStep));
+        return new Recipe(this.uuid.getId(), this.name, this.ingredientList, stepCopy);
+    }
+
+    /**
+     * Changes the specified recipe step with a new step number.
+     */
+    public Recipe modifyRecipeStep(int stepNumber, int newStepNumber) {
+        if (stepNumber > recipeSteps.size()) {
+            throw new IllegalArgumentException("Specified step number cannot exceed the current steps of recipe");
+        }
+        List<RecipeStep> stepCopy = new ArrayList<>();
+        stepCopy.addAll(this.recipeSteps);
+        stepCopy.set(stepNumber - 1, stepCopy.get(stepNumber - 1).modifyStep(newStepNumber));
+        return new Recipe(this.uuid.getId(), this.name, this.ingredientList, stepCopy);
+    }
+
+    /**
+     * Returns the full recipe.
+     */
+    public String getFullRecipe() {
+        StringBuilder ingredients = new StringBuilder();
+        for (Ingredient ingredient : ingredientList) {
+            ingredients.append(ingredient.getName()).append(" ").append(ingredient.getQuantity()).append("\n");
```

Comment:
Just nitpicking, but can probably move each append to a diff line to improve readability